modal/functions.py

```diff
@@ -1162,26 +1162,25 @@ async def _map(self, input_stream: AsyncIterable[Any], order_outputs: bool, retu
         count_update_callback = (
             self._output_mgr.function_progress_callback(self._function_name, total=None) if self._output_mgr else None
         )
-
-        async for item in _map_invocation(
+        it = _map_invocation(
             self.object_id,
             input_stream,
             kwargs,
             self._client,
             order_outputs,
             return_exceptions,
             count_update_callback,
-        ):
-            yield item
+        )
+
+        while 1:
```

Comment:
I *think* this should roughly be equivalent to `async for item in _map_invocation(...)`? Let me know if I missed something here?